File: Colaveco/JavaModel/mobile/Android/gxcommon/com/colaveco/workwithdevicesresultado_resultado_list.java
/*
               File: WorkWithDevicesResultado_Resultado_List
        Description: WorkWithDevicesResultado_Resultado_List
             Author: GeneXus Android Generator version 17_0_7-156546
       Generated on: November 14, 2022 8:16:54.58
       Program type: Callable routine
          Main DBMS: SQLite
*/
package com.colaveco ;
import com.colaveco.*;
import com.genexus.*;
import com.genexus.db.*;
import java.sql.*;
import com.artech.base.services.*;

public final  class workwithdevicesresultado_resultado_list extends GXProcedure implements IGxProcedure
{
   public boolean execute( IPropertiesObject androidPropertiesObject )
   {

      long aP0 = 0;
      long aP1 = 0;
      int aP2 = 0;
      com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt[] aP3 = new com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt[] {new com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt()};

      aP0 = (long) GXutil.lval( androidPropertiesObject.optStringProperty("ExternalId"));
      aP1 = (long) GXutil.lval( androidPropertiesObject.optStringProperty("TipoinformeId"));
      aP2 = (int) GXutil.lval( androidPropertiesObject.optStringProperty("gxid"));

      execute(aP0, aP1, aP2, aP3);

      IEntity outObjGXM1WorkWithDevicesResultado_Resultado_ListSdt =  AndroidContext.ApplicationContext.createEntity("", "WorkWithDevicesResultado_Resultado_List", null);
      if (aP3[0] != null)
      {
         aP3[0].sdttoentity(outObjGXM1WorkWithDevicesResultado_Resultado_ListSdt );
      }
      androidPropertiesObject.setProperty("Gx_Output",outObjGXM1WorkWithDevicesResultado_Resultado_ListSdt);


      return true ;
   }

   public workwithdevicesresultado_resultado_list( int remoteHandle )
   {
      super( remoteHandle , new ModelContext( workwithdevicesresultado_resultado_list.class ), "" );
   }

   public workwithdevicesresultado_resultado_list( int remoteHandle ,
                                                   ModelContext context )
   {
      super( remoteHandle , context, "" );
   }

   public com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt executeUdp( long aP0 ,
                                                                                 long aP1 ,
                                                                                 int aP2 )
   {
      workwithdevicesresultado_resultado_list.this.AV7ExternalId = aP0;
      workwithdevicesresultado_resultado_list.this.AV8TipoinformeId = aP1;
      workwithdevicesresultado_resultado_list.this.AV17gxid = aP2;
      workwithdevicesresultado_resultado_list.this.aP3 = new com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt[] {new com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt()};
      initialize();
      privateExecute();
      return aP3[0];
   }

   public void execute( long aP0 ,
                        long aP1 ,
                        int aP2 ,
                        com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt[] aP3 )
   {
      execute_int(aP0, aP1, aP2, aP3);
   }

   private void execute_int( long aP0 ,
                             long aP1 ,
                             int aP2 ,
                             com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt[] aP3 )
   {
      workwithdevicesresultado_resultado_list.this.AV7ExternalId = aP0;
      workwithdevicesresultado_resultado_list.this.AV8TipoinformeId = aP1;
      workwithdevicesresultado_resultado_list.this.AV17gxid = aP2;
      workwithdevicesresultado_resultado_list.this.aP3 = aP3;
      initialize();
      /* GeneXus formulas */
      /* Output device settings */
      privateExecute();
   }

   private void privateExecute( )
   {
      Gxids = "gxid_" + GXutil.str( AV17gxid, 8, 0) ;
      if ( GXutil.strcmp(Gxwebsession.getValue(Gxids), "") == 0 )
      {
         Gxwebsession.setValue(Gxids+"gxvar_Imagen", AV23Imagen_GXI);
         Gxwebsession.setValue(Gxids+"gxvar_Estado", AV5estado);
         Gxwebsession.setValue(Gxids, "true");
      }
      cleanup();
   }

   protected void cleanup( )
   {
      this.aP3[0] = workwithdevicesresultado_resultado_list.this.AV20GXM1WorkWithDevicesResultado_Resultado_ListSdt;
      CloseOpenCursors();
      exitApplication();
   }

   protected void CloseOpenCursors( )
   {
   }

   /* Aggregate/select formulas */
   public void initialize( )
   {
      AV20GXM1WorkWithDevicesResultado_Resultado_ListSdt = new com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt(remoteHandle, context);
      Gxids = "" ;
      Gxwebsession = AndroidContext.ApplicationContext.getAndroidSession();
      AV23Imagen_GXI = "" ;
      AV5estado = "" ;
      /* GeneXus formulas. */
      Gx_err = (short)(0) ;
   }

   private short Gx_err ;
   private int AV17gxid ;
   private long AV7ExternalId ;
   private long AV8TipoinformeId ;
   private String Gxids ;
   private String AV23Imagen_GXI ;
   private String AV5estado ;
   private com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt[] aP3 ;
   private com.artech.base.services.IAndroidSession Gxwebsession ;
   private com.colaveco.SdtWorkWithDevicesResultado_Resultado_ListSdt AV20GXM1WorkWithDevicesResultado_Resultado_ListSdt ;
}

File: Colaveco/JavaModel/mobile/iOS/gxcommon/workwithdevicesresultado_resultado_list.swift
/*
               File: WorkWithDevicesResultado_Resultado_List
        Description: WorkWithDevicesResultado_Resultado_List
             Author: GeneXus Swift Generator version 17_0_7-156546
       Generated on: November 14, 2022 8:16:56.4
       Program type: Callable routine
          Main DBMS: SQLite
*/

@objc( workwithdevicesresultado_resultado_list )
public final  class workwithdevicesresultado_resultado_list: GXProcedure
{
   @objc public func executeFlexibleClient( _ aP0: NSNumber? ,
                                            _ aP1: NSNumber? ,
                                            _ aP2: NSNumber? ,
                                            _ inout_aP3: AutoreleasingUnsafeMutablePointer<SdtWorkWithDevicesResultado_Resultado_ListSdt?> )
   {
      let aP3: SdtWorkWithDevicesResultado_Resultado_ListSdt? = inout_aP3.pointee
      let iP0: Int64 = (aP0 == nil) ? 0 : aP0!.int64Value as Int64
      let iP1: Int64 = (aP1 == nil) ? 0 : aP1!.int64Value as Int64
      let iP2: Int = (aP2 == nil) ? 0 : aP2!.intValue as Int
      var iP3: SdtWorkWithDevicesResultado_Resultado_ListSdt = (aP3 == nil) ? SdtWorkWithDevicesResultado_Resultado_ListSdt() : aP3! as SdtWorkWithDevicesResultado_Resultado_ListSdt
      execute(iP0, iP1, iP2, &iP3)
      inout_aP3.pointee  = iP3 as SdtWorkWithDevicesResultado_Resultado_ListSdt?
   }

   public func executeUdp( _ aP0: Int64 ,
                           _ aP1: Int64 ,
                           _ aP2: Int ) -> SdtWorkWithDevicesResultado_Resultado_ListSdt
   {
      self.AV7ExternalId = aP0
      self.AV8TipoinformeId = aP1
      self.AV17gxid = aP2
      initialize()
      autoreleasepool
      {
         privateExecute()
      }
      return self.AV20GXM1WorkWithDevicesResultado_Resultado_ListSdt
   }

   public func execute( _ aP0: Int64 ,
                        _ aP1: Int64 ,
                        _ aP2: Int ,
                        _ aP3: inout SdtWorkWithDevicesResultado_Resultado_ListSdt )
   {
      execute_int(aP0, aP1, aP2, &aP3)
   }

   private func execute_int( _ aP0: Int64 ,
                             _ aP1: Int64 ,
                             _ aP2: Int ,
                             _ aP3: inout SdtWorkWithDevicesResultado_Resultado_ListSdt )
   {
      self.AV7ExternalId = aP0
      self.AV8TipoinformeId = aP1
      self.AV17gxid = aP2
      initialize()
      /* GeneXus formulas */
      /* Output device settings */
      autoreleasepool
      {
         privateExecute()
      }
      aP3 = self.AV20GXM1WorkWithDevicesResultado_Resultado_ListSdt
   }

   private func privateExecute( )
   {
      Gxids = "gxid_"  +  String.gxStringFromNumber(Decimal(AV17gxid), totalLength:8, decimals:0)
      if ( GXutil.strcmp(Gxwebsession.getValue(Gxids), "") == 0 )
      {
         Gxwebsession.setValue(Gxids+"gxvar_Imagen", AV23Imagen_GXI)
         Gxwebsession.setValue(Gxids+"gxvar_Estado", AV5estado)
         Gxwebsession.setValue(Gxids, "true")
      }
      cleanup()
   }

   public override func cleanup( )
   {
      closeOpenCursors()
   }

   private func closeOpenCursors( )
   {
   }

   /* Aggregate/select formulas */
   public override func initialize( )
   {
      AV20GXM1WorkWithDevicesResultado_Resultado_ListSdt = SdtWorkWithDevicesResultado_Resultado_ListSdt()
      Gxids = ""
      Gxwebsession = GXOfflineSession.sharedInstance()
      AV23Imagen_GXI = ""
      AV5estado = ""
      /* GeneXus formulas. */
      Gx_err = 0
   }

   private var AV17gxid : Int = 0
   private var Gx_err : Int = 0
   private var AV7ExternalId : Int64 = 0
   private var AV8TipoinformeId : Int64 = 0
   private var Gxids : String = ""
   private var AV23Imagen_GXI : String = ""
   private var AV5estado : String = ""
   private var Gxwebsession : GXOfflineSession = GXOfflineSession()
   private var AV20GXM1WorkWithDevicesResultado_Resultado_ListSdt : SdtWorkWithDevicesResultado_Resultado_ListSdt = SdtWorkWithDevicesResultado_Resultado_ListSdt()
}

